server/eclipse-project/src/main/java/us/freeandfair/corla/asm/AbstractStateMachine.java

```diff
@@ -217,6 +228,16 @@ public ASMState stepTransition(final ASMTransition the_transition)
    */
   public ASMState stepEvent(final ASMEvent the_event)
       throws IllegalStateException {
+    // refresh and skip events are always permitted
```

Comment:
Eek. I don't like having these in the AbstractStateMachine class directly. Is there a way that we can annotate the events in the enumeration instead (like a method `isAlwaysEnabled()`), or throw them in a collection of always-enabled events that can be added to by child classes, or something?